consensus/safety-rules/src/safety_rules.rs

```diff
@@ -56,6 +57,16 @@ impl<T: Payload> SafetyRules<T> {
         }
     }
 
+    fn signer(&self) -> Result<&ValidatorSigner, Error> {
+        self.validator_signer.as_ref().ok_or(Error::NotInitialized)
```

Comment:
what's not initialized, we should probably make this take in a string to specify which field it failed on